src/modules/SourceDialogs.rb

```diff
@@ -2713,6 +2713,19 @@ def valid_scheme?(url)
     publish :function => :TypePopup, :type => "string ()"
     publish :function => :TypeDialog, :type => "symbol ()"
     publish :function => :TypeDialogDownloadOpt, :type => "map <string, any> ()"
+
+  private
+
+    # Build URL from tokens
+    # @param [Hash] with keys:"scheme","http","host","port","path","query",...
+    # @return [String] url, empty string if invalid data is used to build the url.
+    def build_url(url_settings)
+      # URL.Build is escaping characters (e.g. ":") which is wrong for
+      # pathes like /dist/ibs/SUSE:/SLE-12-SP1:/GA. (bnc#966413)
+      # So we are unescaping it.
+      URLRecode.UnEscape(URL.Build(url_settings))
```

Comment:
this looks wrong, as it also unescape other characters that are correctly escaped and cannot be in URL. I think we really need testcases for this and be sure that we return valid url, as this can seriously break installation. maybe only escaping `:` is wrong?